dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -136,6 +139,35 @@ abstract class Recorder {
   /// This is useful for doing one-time clean up work after the benchmark is
   /// complete.
   Future<void> tearDownAll() async {}
+
+  /// Whether we are recording a measured frame.
+  ///
+  /// This flag ensures that [endMeasureFrame] is called only
+  /// when a [startMeasureFrame] is called earlier.
+  /// Because we want to skip warm-up frames, this flag is necessary.
+  bool _isMeasuringFrame = false;
+
+  /// Start recording a measured frame if we are not in the warm-up phase.
+  ///
+  /// Use [startMeasureFrameSkippingWarmUp] and
+  /// [endMeasureFrameSkippingWarmUp] to skip measuring warm-up frames.
+  void startMeasureFrameSkippingWarmUp() {
+    if (!isWarmingUp) {
+      startMeasureFrame();
+      _isMeasuringFrame = true;
+    }
+  }
+
+  /// Stop recording a measured frame, if any recording is in process.
+  ///
+  /// Use [startMeasureFrameSkippingWarmUp] and
+  /// [endMeasureFrameSkippingWarmUp] to skip measuring warm-up frames.
+  void endMeasureFrameSkippingWarmUp() {
+    if (_isMeasuringFrame) {
```

Comment:
If we reach this line and `_isMeasuringFrame` is false, doesn't it indicate that there's a bug in the benchmark or in the recorder? If so, we should probably throw an error rather than silently keep on trucking.

Reply:
This pattern takes care of the following scenario:

If the `isWarmingUp` flag is set to `false` when a frame is **being** measured, then the `startMeasureFrameSkippingWarmUp` method, which was called **before** the switch, would not start measuring a frame. So, the `endMeasureFrameSkippingWarmUp` method, which is going to be called **after** the switch, should not call `endMeasureFrame` either.

If `_isMeasuringFrame` is false, this means we were still in the warm-up phase when the proposed measured frame starts, so we should not measure this frame.

If we just decide whether to call `startMeasureFrame` and `endMeasureFrame` based on whether we are **currently** in a warm-up phase, then in the scenario above, `endMeasureFrame` would be called without a `startMeasureFrame` call first, causing an error.

So, it's to make sure that `endMeasureFrame` is called only when a `startMeasureFrame` has been called earlier.

Reply:
Modified according to discussion.